core/server/utils/blog-icon.js

```diff
@@ -0,0 +1,102 @@
+var ICO = require('icojs'),
+    errors = require('../errors'),
+    url = require('./url'),
+    Promise = require('bluebird'),
+    i18n = require('../i18n'),
+    settingsCache = require('../settings/cache'),
+    fs = require('fs'),
+    _ = require('lodash'),
+    getIconDimensions,
+    isIcoImageType,
+    getIconType,
+    getIconUrl;
+
+/**
+ * Get dimensions for ico file from its real file storage path
+ * Always returns {object} getIconDimensions
+ * @param {string} path
+ * @returns {Promise<Object>} getIconDimensions
+ * @description Takes a file path and returns ico width and height.
+ */
+getIconDimensions = function getIconDimensions(path) {
+    return new Promise(function getIconSize(resolve, reject) {
+        var arrayBuffer;
+
+        arrayBuffer = new Uint8Array(fs.readFileSync(path)).buffer;
+        ICO.parse(arrayBuffer).then(function (result, error) {
```

Comment:
same here. we need to catch the error. having the error as second parameter on `.then` won't work.